### PR TITLE
[Cortal Consors] Add support for "new" broker

### LINKED ISSUE #228: [Consorsbank] Ein Kauf kann nicht importiert werden
## Document
<details><summary>KAUF_838891405_ord77830814_001_wkn251124_dat20140217_id382569190.pdf</summary><p>

```json
[
  [
    "Cortal Consors S.A. Zweigniederlassung Deutschland",
    "Bahnhofstraße 55, 90402 Nürnberg, HR Nürnberg B20075, USt-IdNr. DE225900761, info@cortalconsors.de, www.cortalconsors.de",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00",
    "Sitz der Cortal Consors S.A.: 1, boulevard Haussmann, 75318 Paris CEDEX 09, Frankreich, Registergericht: R.C.S. Paris 327 787 909",
    "Président du Conseil d’Administration (Verwaltungsratsvorsitzender) und Directeur Général (Generaldirektor) der Cortal Consors S.A.: Olivier Le Grand",
    "Leitung der Zweigniederlassung Deutschland: Kai Friedrich (CEO), Richard Döppmann, Stefan Gröning, Dr. Gérard Derszteler",
    "Cortal Consors • 90318 Nürnberg",
    "WERTPAPIERABRECHNUNG",
    "KAUF",
    "AM 17.02.2014",
    "UM",
    "08:02:48",
    "MUENCHEN",
    "NR. 77830814.001",
    "Depotnummer",
    "5136356671",
    "0409955449/32",
    "Vermerk der Bank",
    "1000",
    "00",
    "Duvpf Jbjnfxoor",
    "Uaqhlgbihdgq. 0",
    "49634 Jüllcwrj",
    "Hinweis für Kontoauszüge, Wertpapierabrechnungen und Depotbuchungsanzeigen:",
    "Kapitalerträge und Spekulationsgewinne sind einkommensteuerpflichtig.",
    "Einwendungen wegen Unrichtigkeit oder Unvollständigkeit dieser Mitteilung oder die Nichtgenehmigung einer im Wege der Einzugsermächtigung erfolgten",
    "Lastschriftabbuchung müssen unverzüglich erhoben werden, vgl. Nummer B. Ziffer I. 11 (4) und (5) der Allgemeine Geschäftsbedingungen (AGB Banken) sowie",
    "unter B. VII. Ziffer 2.4 der Lastschriftbedingungen. Umsätze und Kontobuchungen, die nach dem Erstellungsdatum anfallen und sich auf den Abrechnungssaldo des",
    "abgelaufenen Abrechnungszeitraumes auswirken, werden erst mit dem folgenden Kontoauszug ausgewiesen. Korrekturen werden seitens der Bank gekennzeichnet.",
    "Machen Sie Ihre Einwendungen in Textform geltend, genügt die Absendung innerhalb der Sechs-Wochen-Frist an Cortal Consors Deutschland (Revision) oder per",
    "Fax oder Mail an die unten angegebenen Adressen. Das Unterlassen rechtzeitiger Einwendungen gilt als Genehmigung.",
    "Wertpapier",
    "WKN",
    "ISIN",
    "ISHS-EO CO.BD LA.C.UTS DZ",
    "251124",
    "DE0002511243",
    "Einheit",
    "Umsatz",
    "ST",
    "0,38007",
    "Kurs",
    "129,610000",
    "EUR",
    "P.ST.",
    "Kurswert",
    "EUR",
    "49,26",
    "Provision",
    "EUR",
    "0,74",
    "Wert 19.02.2014",
    "EUR",
    "50,00",
    "zulasten Konto-Nr. 0833891400",
    "Wertpapiere zugunsten Girosammelverwahrung",
    "Clearstream Bk Ffm 2126",
    "Die Order wurde an folgender Börse gehandelt: MUENCHEN",
    "Sparplanname: Euro Anleihen"
  ],
  [
    "Cortal Consors S.A. Zweigniederlassung Deutschland",
    "Bahnhofstraße 55, 90402 Nürnberg, HR Nürnberg B20075, USt-IdNr. DE225900761, info@cortalconsors.de, www.cortalconsors.de",
    "Fon +49 (0) 911 / 369-30 00, Fax +49 (0) 911 / 369-10 00",
    "Sitz der Cortal Consors S.A.: 1, boulevard Haussmann, 75318 Paris CEDEX 09, Frankreich, Registergericht: R.C.S. Paris 327 787 909",
    "Président du Conseil d’Administration (Verwaltungsratsvorsitzender) und Directeur Général (Generaldirektor) der Cortal Consors S.A.: Olivier Le Grand",
    "Leitung der Zweigniederlassung Deutschland: Kai Friedrich (CEO), Richard Döppmann, Stefan Gröning, Dr. Gérard Derszteler",
    "Infos bei Käufen bzw. Verlust bei Wertpapieren ungleich Aktien",
    "Gezahlter Zwischengewinn",
    "EUR",
    "0,68",
    "In den Verrechnungstopf Allgemein eingestellt",
    "EUR",
    "0,68",
    "Summe ergibt",
    "KAPST-pflichtige Kapitalerträge",
    "EUR",
    "0,00",
    "Mit Sparerpauschbetrag verrechnet",
    "EUR",
    "0,00",
    "Bemessungsgrundlage für KAPST vor QUST",
    "EUR",
    "0,00",
    "An Kapitalertrag anrechenbare. ausl. QUST 0,00*4",
    "EUR",
    "0,00",
    "Bemessungsgrundlage für KAPST",
    "EUR",
    "0,00",
    "Bemessungsgrundlage für KAPST anteilig 100,00%",
    "EUR",
    "0,00",
    "Salden nach dem Geschäft",
    "Verrechnungstopf Aktien nach dem Geschäft",
    "EUR",
    "0,00",
    "Verrechnungstopf Allg. nach dem Geschäft",
    "EUR",
    "121,58",
    "Verrechnungstopf QUST nach dem Geschäft",
    "EUR",
    "0,00",
    "Sparerpauschbetrag nach dem",
    "Geschäft",
    "EUR",
    "0,00",
    "Infos zu Investmentpreisen",
    "Akkum. ausschüttungsgleiche Erträge",
    "EUR",
    "1,54",
    "Bereinigte akkum. ausschüttungsgleiche Erträge",
    "EUR",
    "1,52",
    "Akkum. Substanzausschüttungen",
    "EUR",
    "0,06",
    "Seite 2 zu Wertpapierabrechnung Nr.",
    "77830814.001 vom",
    "17.02.2014",
    "Depotnummer 0838891405",
    "Hinweis für Kontoauszüge, Wertpapierabrechnungen und Depotbuchungsanzeigen:",
    "Kapitalerträge und Spekulationsgewinne sind einkommensteuerpflichtig.",
    "Einwendungen wegen Unrichtigkeit oder Unvollständigkeit dieser Mitteilung oder die Nichtgenehmigung einer im Wege der Einzugsermächtigung erfolgten",
    "Lastschriftabbuchung müssen unverzüglich erhoben werden, vgl. Nummer B. Ziffer I. 11 (4) und (5) der Allgemeine Geschäftsbedingungen (AGB Banken) sowie",
    "unter B. VII. Ziffer 2.4 der Lastschriftbedingungen. Umsätze und Kontobuchungen, die nach dem Erstellungsdatum anfallen und sich auf den Abrechnungssaldo des",
    "abgelaufenen Abrechnungszeitraumes auswirken, werden erst mit dem folgenden Kontoauszug ausgewiesen. Korrekturen werden seitens der Bank gekennzeichnet.",
    "Machen Sie Ihre Einwendungen in Textform geltend, genügt die Absendung innerhalb der Sechs-Wochen-Frist an Cortal Consors Deutschland (Revision) oder per",
    "Fax oder Mail an die unten angegebenen Adressen. Das Unterlassen rechtzeitiger Einwendungen gilt als Genehmigung."
  ]
]

```

</p></details>

---
Issue submitted with https://import.tresor.one

### PR DESCRIPTION
**Pull request by dredav**
- This PR replace the old one #77
- Fix #228, #230, #232, #301
- Add test cases
- Update test structure and test cases
- Use `validateActivity` function, dividend payout use now the gross amount, dividend date now set to pay date
- Parse `Provision` as fee
- Add test case and add support for etf dividends
- Add church tax parsing
- eslint format in merge 🙄 
- **required the merge of #249 before mergin this PR**

### CODE REVIEW

File: tests/brokers/__mocks__/cortalconsors/sell/2014_allianz.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
+[
+  [
+    "BNP Paribas S.A. Niederlassung Deutschland",
```
Comment (SirGibihm) [inline tests/brokers/__mocks__/cortalconsors/sell/2014_allianz.json:3]: Yet another BNP Paribas Broker. There might be a lot of code duplication

File: tests/brokers/cortalconsors.test.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+  describe('Buy', () => {
+    test('Should map the document correctly: 2014_allianz', () => {
+      const activities = cortalconsors.parsePages(buySamples[0]).activities;
+
+      expect(activities).toEqual([
+        {
+          broker: 'cortalconsors',
+          type: 'Buy',
+          date: '2014-03-07',
```
Comment (SirGibihm) [inline tests/brokers/cortalconsors.test.js:40]: Datetime is not supported by can be extracted from the .pdf. Could you add that functionality please?

File: src/brokers/cortalconsors.js [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+  }
+
+  return content[lineNumber + 1].split('WKN:').pop().trim();
+};
+
+const findDateBuySell = textArr => {
+  const idx = textArr.findIndex(
+    t => t.toLowerCase() === 'wertpapierabrechnung'
+  );
+  const date = textArr[idx + 2].substr(3, 10).trim();
```
Comment (SirGibihm) [inline src/brokers/cortalconsors.js:46]: You don't use the function that helps you with parsing the time from the helper class, is this on purpose?

File: src/brokers/cortalconsors.js [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+  return date;
+};
+
+const findDateDividend = content => {
+  const line = content[content.findIndex(line => line.startsWith('WERT'))];
+  return line.split(/\s+/)[1];
```
Comment (SirGibihm) [inline src/brokers/cortalconsors.js:53]: Same here with the date function

**Review by SirGibihm — CHANGES_REQUESTED** [review]
Sligth changes requested. Sadly, the example documents don't allow for the implementation of FxRates etc. to be implemented as of yet. Furthermore, could you check wether the logic of this broker is the same as for smartbroker and onvista? I can't code too much atm myself so I could not check it.

EDIT: Merge conflicts, gotta love ESLINT 👯

**Comment by dredav** [comment]
@SirGibihm hmm yes. I dismissed the datetime logic becase of doing a pull request against the master branch and the datetime logic is not merged yes - but we need to do this. So I can rebase this feature again my datetime PR and we need to merge that feature first and then this one.

File: tests/brokers/__mocks__/cortalconsors/sell/2014_allianz.json (hunk 1 above)
Comment (dredav) [reply tests/brokers/__mocks__/cortalconsors/sell/2014_allianz.json:3]: I don't compare all functions, but the first functions I compared has quite different content.

**Review by SirGibihm — APPROVED** [review]

**Pull request merged**